backend/app/controllers/system_messages.js

```diff
@@ -9,49 +8,14 @@ module.exports = {
         ) {
             res.status(400).send('System message must have a \'to\' field and a \'message\' field').end();
         } else {
-            messageService.sendSystemMessage(
-                req.app.get(messageService.SYSTEM_MESSAGE_USER_TOKEN_FIELD),
-                req.body.to,
-                req.body.message,
-                messageService.SYSTEM_MESSAGE_SUBJECT
-            )
-            .then(() => {
+            const response = yield * common.sendSystemMessageWithMessageService(req, req.body.to, req.body.message);
```

Comment:
This `yield` is undefined, giving a 400 `{
  "message": "yield is not defined"
}` in response to the endpoint. I think we need a `*` on this function declaration